src/framework/components/element/component.js

```diff
@@ -142,33 +283,67 @@ pc.extend(pc, function () {
                     child = this._children[i];
                     child.dirtyWorld = true;
                     child._aabbVer++;
-
                 }
+
+                this.element.fire("resize", this.element._width, this.element._height);
+            }
+        },
+
+        _drawDebugBox: function(dt) {
+            var bottomLeft = new pc.Vec3();
```

Comment:
In such case, it would be better to put this under #define DEBUG, so it does not gets into production engine code, and debugging to be non-public method. Devs tend to use everything out there, and if it is "official" API, they will expect it to perform, as their use cases might surprise on daily basis, which easily leads to big performance issues :)